test/test_overrides.py

```diff
@@ -0,0 +1,756 @@
+import torch
+import numpy as np
+import unittest
+import six
+import inspect
+import pprint
+
+from common_utils import TestCase
+
+HANDLED_FUNCTIONS = {}
```

Comment:
I think it's important to more clearly document how the `HANDLED_FUNCTIONS` pattern works, because I think there is a decent chance that people who are actually going to use `__torch_function__` will structure their code in a similar way.

Also cc @cpuhrsch, how does this API look to you, if you wanted to do nested tensor in Python?

Reply:
Agreed, I'll add comments here explaining how this works but we're also planning to add documentation to this pull request before merging where this will be explained in a user-facing manner.